File: python/to_lower_case.py
class Solution:
    def isAlphabet(self, character: str) -> bool:
        return self.isUpperCase(character) or self.isLowerCase(character)

    def isUpperCase(self, character: str) -> bool:
        return 65 <= ord(character) <= 90

    def isLowerCase(self, character: str) -> bool:
        return 97 <= ord(character) <= 122

    def toLowerCase(self, s: str) -> str:
        result = ''
        for character in s:
            result += chr(ord(character) + 32) if self.isAlphabet(character) and self.isUpperCase(character) else character
        return result

File: src/ToLowerCase.java
public class ToLowerCase {
    public String toLowerCase(String s) {
        StringBuilder result = new StringBuilder();
        for (int index = 0 ; index < s.length() ; index++) {
            result.append(toLowerCase(s.charAt(index)));
        }
        return result.toString();
    }

    private char toLowerCase(char character) {
        return isUpperCase(character) && isAlphabet(character) ? (char) (character + 32) : character;
    }

    private boolean isAlphabet(char character) {
        return isUpperCase(character) || isLowerCase(character);
    }

    private boolean isLowerCase(char character) {
        return 97 <= character && character <= 122;
    }

    private boolean isUpperCase(char character) {
        return 65 <= character && character <= 90;
    }
}
